### PR TITLE
cas: elaborate error from os.Lstat in visitSymlink

### PR DESCRIPTION
**Pull request by atetubou**
This is to debug https://crbug.com/1237607

### CODE REVIEW

File: go/pkg/cas/upload.go [hunk 1]
```diff
@@ -395,7 +395,13 @@ func (c *Client) Upload(ctx context.Context, opt UploadOptions, inputC <-chan *U
 		}
 	})
 
-	return &UploadResult{Stats: u.stats, u: u}, errors.WithStack(eg.Wait())
+	err := eg.Wait()
+	if err != nil {
+		// Caller may not unwrap stack frame of github.com/pkg/error, so log error here.
+		log.Errorf("upload err %+v", err)
```
Comment (rubensf) [inline go/pkg/cas/upload.go:401]: Maybe a more descriptive message? If only for a client grepping for error code to find out easier :) eg "failed to upload at least one blob: %+v"

**Review by rubensf — APPROVED** [review]

File: go/pkg/cas/upload.go [hunk 2]
```diff
@@ -395,7 +395,13 @@ func (c *Client) Upload(ctx context.Context, opt UploadOptions, inputC <-chan *U
...
+	}
+
+	return &UploadResult{Stats: u.stats, u: u}, err
```
Comment (nodirg) [inline go/pkg/cas/upload.go:404]: Why not have the callsite log the error? Modifying a common library seems more intrusive than the less global callsite.

**Review by nodirg — APPROVED** [review]

File: go/pkg/cas/upload.go (hunk 2 above)
Comment (atetubou) [reply go/pkg/cas/upload.go:404]: Are you suggesting to support stack extraction from github.com/pkg/errors in go.chromium.org/luci/common/errors?

File: go/pkg/cas/upload.go (hunk 1 above)
Comment (atetubou) [reply go/pkg/cas/upload.go]: done

**Review by Jwata — APPROVED** [review]

File: go/pkg/cas/upload.go (hunk 2 above)
Comment (nodirg) [reply go/pkg/cas/upload.go]: no? I mean, this code returns the exactly same error object that it logs. It means the callsite can do the same.

Comment (nodirg) [reply go/pkg/cas/upload.go]: I see the latest version now returns an error with a stack annotation.
I am not sure what's the problem logging with stack, but perhaps it does not matter much.

Comment (atetubou) [reply go/pkg/cas/upload.go]: I understood what you mean. Thanks.

**Pull request merged**